controllers/mpicluster_controller.go

```diff
@@ -16,6 +17,7 @@ import (
 func MPICluster(mgr ctrl.Manager, webhooksEnabled, istioEnabled bool) error {
 	reconciler := core.NewReconciler(mgr).
 		For(&dcv1alpha1.MPICluster{}).
+		Component("istio-peerauthentication", mpi.IstioPeerAuthentication(istioEnabled)).
```

Comment:
Sadly this will need to be updated from your other refactor. Sorry for the delay in reviewing.

Reply:
Yeah, it'll need a slight edit to be merged.